From b5607dd4c8e7f5df8579e2aa87aa341f0cb005f5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Fri, 14 Mar 2025 11:04:57 +0100
Subject: [PATCH 876/954] fix(st-bsec): update bsec state to match other
 components
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Needed otherwise BSEC mirror is not synchronized with TF-M.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9fcdce1ced2274f89aae19ec30b15c3cc3bf9dbb fix(st-bsec): update bsec state to match other components]
Change-Id: Iba98d2e5b5f8cfbff2620a57952504c3c9fa183c
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/444227
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 include/drivers/st/bsec.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/drivers/st/bsec.h b/include/drivers/st/bsec.h
index 00af06bdcc..7999f7d3dd 100644
--- a/include/drivers/st/bsec.h
+++ b/include/drivers/st/bsec.h
@@ -31,13 +31,13 @@
  * BSEC open is considered invalid in OSTL use case.
  * @state: global state
  *           [1:0] BSEC state
- *             0b: Closed
- *             1b: Invalid
+ *             0Xb: Closed
+ *             1Xb: Invalid
  *           [8]: Hardware Key set = 1b
  */
 #define BSEC_STATE_CLOSED		U(0x0)
-#define BSEC_STATE_INVALID		U(0x1)
-#define BSEC_STATE_MASK			BIT(0)
+#define BSEC_STATE_INVALID		U(0x2)
+#define BSEC_STATE_MASK		        BIT(1)
 
 #define BSEC_HARDWARE_KEY		BIT(8)
 
-- 
2.53.0

